versioned_docs/version-5.x/configuring-links.md

```diff
@@ -0,0 +1,355 @@
+      Notifications: 'notify/:user',
+    },
+  },
+};
+```
+
+Then, the path `/users/42` will resolve to the following state object:
+
+```js
+const state = {
+  routes: [
+    {
+      name: 'Home',
+      state: {
+        index: 1,
+        routes: [
+          {
+            name: 'Notifications',
+          },
+          {
+            name: 'Profile',
+            params: { id: 'jane' },
+          },
+        ],
+      },
+    },
+  ],
+};
+```
+
+Notice that we can't pass any params to the `Notifications` screen if it isn't explicitly mentioned in the URL string, so the screen should implement handling lack of these params with e.g. providing default ones.
+
+For some advanced cases, specifying the mapping may not be sufficient. You can implement your custom parser to address these cases using the `getStateFromPath` option:
+
+```js
+const linking = {
+  prefixes: ['https://mychat.com', 'mychat://'],
+  config: {
+    Chat: 'feed/:sort',
+  },
+  getStateFromPath: (path, options) => {
+    // Implement your custom parser and return the navigation state here
+  },
+};
+```
+
+## Example App
+
+### Overview
+
+In the example app, we will use the Expo managed workflow. The guide will focus on creating the deep linking configuration and not on creating the components themselves, but you can always check the full implementation in the [github repo](https://github.com/WoLewicki/DeepLinkingExample).
+
+### Getting Started
+
+In the beginning, we need to create a structure for our application. To keep it simple, the main navigator will be bottom-tabs navigator with two screens. Its first screen will be a simple stack navigator, called `HomeStack`, with two screens: `Home` and `Profile`, and the second tabs screen will be just a simple one without any nested navigators, called `Settings`. After creating the app's structure, we can prepare a deep linking config object, which will statically recreate the structure with a URL path matching every state we would like to achieve. In our example, it would perhaps look like this:
+
+```js
+config = {
+  HomeStack: 'stack',
+  Home: 'home',
+  Profile: 'user',
+  Settings: 'settings',
+};
+```
+
+Then, to get to `Home` screen in `HomeStack`, the URL would look like `/stack/home`, and for `Settings` screen, it would be `/settings`.
+For such an app, without any parameters added to the routes and a very simple structure, it doesn't make too much sense to explicitly create `config`. If it wasn't present, the routes would be parsed by their names, so the first URL would have to be `/HomeStack/Home` and the second one `/Settings`.
+
+But what if we want to have parameters in the URL, like the name of the person, whose profile we are visiting in the `Profile` screen? Or maybe we would like to define a screen that will be always present in the state for the `HomeStack` navigator? With properly defined `config` it can all be done easily, so let's do this!
+
+### Defining better config
```

Comment:
we should be more precise about what makes this config better

Reply:
So maybe "Defining extended config"?